src/index.js

```diff
@@ -26,7 +37,18 @@ function Block (data) {
       hashFunc = 'sha2-256'
     }
 
-    multihashing(this.data, hashFunc, callback)
+    if (this._cache[hashFunc]) {
+      setImmediate(() => {
```

Comment:
`setImmediate` is also used in 4 different places across the codebase. There is also polyfill for it if it comes to it.